.github/workflows/ci.yml

```diff
@@ -1,62 +1,67 @@
 on:
   push:
-    branches: [ staging, trying, master ]
+    branches: master
   pull_request:
+  merge_group:
 
 name: Continuous Integration
 
 jobs:
+  ci:
+    name: CI
+    runs-on: ubuntu-latest
+    needs: [build, test, test-strict]
+    if: always()
+    steps:
+      - name: Done
+        run: jq --exit-status 'all(.result == "success")' <<< '${{ toJson(needs) }}'
+
   # check if the project builds with MSRV, stable and nighly
   build:
+    name: "Build (${{ matrix.name || matrix.rust }})"
     runs-on: ubuntu-latest
     continue-on-error: ${{ matrix.experimental || false }}
     strategy:
       matrix:
-        rust: [ 1.56.0, stable ]
+        rust: [ stable ]
         include:
           # Test nightly but don't fail the build.
           - rust: nightly
             experimental: true
+          # MSRV
+          - rust: 1.58.0
+            name: "MSRV"
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
       - name: Install Rust
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@master
         with:
           toolchain: ${{ matrix.rust }}
-          profile: minimal
-          override: true
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
+          components: clippy
+      - run: cargo check
+        env:
+          RUSTFLAGS: -D warnings
+      - run: cargo clippy
```

Comment:
this should be fine imo since clippy doesn't add "error-by-default" lints just like that without reason, most new lints are added as warn or even allow